games-engines/openscenegraph/openscenegraph-3.6.0.recipe

```diff
@@ -0,0 +1,196 @@
+SUMMARY="An open source, real-time graphics middle-ware used by developers"
+DESCRIPTION="The OpenSceneGraph is an open source high performance 3D \
+graphics toolkit, used by application developers in fields such as visual \
+simulation, games, virtual reality, scientific visualization and modelling. \
+Written entirely in Standard C++ and OpenGL it runs on all Windows platforms, \
+OSX, GNU/Linux, IRIX, Solaris, HP-Ux, AIX and FreeBSD operating systems. The \
+OpenSceneGraph is now well established as the world leading scene graph \
+technology, used widely in the vis-sim, space, scientific, oil-gas, games and \
+virtual reality industries."
+HOMEPAGE="http://www.openscenegraph.org/"
+COPYRIGHT="2002-2018 Robert Osfield and others"
+LICENSE="OSGPL"
+REVISION="1"
+SOURCE_URI="https://github.com/openscenegraph/OpenSceneGraph/archive/OpenSceneGraph-$portVersion.tar.gz"
+CHECKSUM_SHA256='6f57134ea74a39f1c7b24c285e6278cf906c47f6c681573b86d12173a466efed'
+SOURCE_DIR="OpenSceneGraph-OpenSceneGraph-$portVersion"
+PATCHES="openscenegraph-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion=$portVersion
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	openscenegraph$secondaryArchSuffix = $portVersion
+	lib:libOpenThreads$secondaryArchSuffix = $libVersionCompat
```

Comment:
This library have it's own `SONAME`, so you would want a separate `libOpenThreadsVersionCompat` for it.